### PR TITLE
fix(fa): More detail error messages

### PR DESCRIPTION
**Pull request by MargretFinnboga**
# More detail error messages when creating municipality and staff

https://app.asana.com/0/1201722624788165/1201672092806062

## What

- Catching error when creating staff 
- Handling staff creation error in front-end

## Why

- For more detail error messages

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/financial-aid/web-veita/src/components/NewMunicipalityModal/NewMunicipalityModal.tsx [hunk 1]
```diff
@@ -50,6 +49,10 @@ const NewMunicipalityModal = ({
     hasSubmitError: false,
   })
 
+  const [errorMessage, setErrorMessage] = useState<string>(
+    'Eitthvað fór úrskeiðis, vinsamlega reynið aftur síðar',
```
Comment (Rikkiabb) [inline apps/financial-aid/web-veita/src/components/NewMunicipalityModal/NewMunicipalityModal.tsx:53]: Do you want to have a default value here?

File: apps/financial-aid/backend/src/app/modules/staff/staff.service.ts [hunk 2]
```diff
@@ -139,25 +139,32 @@ export class StaffService {
...
-        email: input.email,
-        roles: input.roles,
-        active: true,
-        municipalityName: municipality.municipalityName,
-        municipalityHomepage: municipality?.municipalityHomepage,
-      },
-      { transaction: t },
-    )
+
+    return newStaff
```
Comment (Rikkiabb) [inline apps/financial-aid/backend/src/app/modules/staff/staff.service.ts:167]: could we rename to staff?

File: apps/financial-aid/backend/src/app/modules/municipality/municipality.service.ts [hunk 3]
```diff
@@ -102,7 +102,9 @@ export class MunicipalityService {
               })
             })
         })
-        .catch((err) => err)
+        .catch((err) => {
```
Comment (davidpxn) [inline apps/financial-aid/backend/src/app/modules/municipality/municipality.service.ts:105]: Can we remove this catch if we are just throwing the error again?

File: apps/financial-aid/web-veita/src/components/NewMunicipalityModal/NewMunicipalityModal.tsx [hunk 4]
```diff
@@ -81,7 +82,15 @@ const NewMunicipalityModal = ({
       }).then(() => {
         onMunicipalityCreated()
       })
-    } catch (e) {
+    } catch (error) {
+      if (error.graphQLErrors[0]?.extensions?.response?.status === 400) {
+        setErrorMessage(
+          'Misstókst að búa til stjórnanda, mögulega er kennitala í notkun',
```
Comment (Rikkiabb) [inline apps/financial-aid/web-veita/src/components/NewMunicipalityModal/NewMunicipalityModal.tsx:88]: Typo here in mistókst

**Review by davidpxn — APPROVED** [review]
Nicee👌

**Pull request merged**